### PR TITLE
[exporter/azuremonitor] Add Connection String Support to Azure Monitor Exporter

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
**Description:** <Describe what has changed.>

This pull request introduces the ability to configure the Azure Monitor Exporter using a connection string, aligning the exporter configuration with Azure Monitor's recommended practices. The current implementation requires users to set the instrumentation key directly, which will soon be deprecated in favor of using the connection string.

**Changes Made:**

1. Configuration Update: Modified the `Config` struct and related configuration parsing logic to support a `ConnectionString` field.
2. Parsing Logic: Implemented functionality to parse the connection string and extract necessary details, such as `InstrumentationKey` and `IngestionEndpoint`.
3. Updated Tests: Revised existing tests and added new ones to ensure coverage of the new configuration option.

**Benefits:**

* Streamlines the configuration process for end-users.
* Aligns with Azure Monitor's best practices and recommended configuration approach.
* Paves the way for the upcoming deprecation of direct instrumentation key configuration.

**Backwards Compatibility:**
This update maintains full backwards compatibility. Users currently utilizing the instrumentation key for configuration can continue to do so but are advised to transition to using the connection string.

**To-Do** 

* Documentation Update in a follow up PR
* Deprecation Notice: A future update will introduce a deprecation warning for users still configuring the exporter with the instrumentation key, encouraging them to switch to using a connection string.
* Add support  for `EndpointSuffix` in connection string - https://learn.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=dotnet5#connection-string-with-an-endpoint-suffix

**Link to tracking Issue:** <Issue number if applicable> 
#28853

**Testing:** <Describe what testing was performed and which tests were added.>

Conducted comprehensive testing, including unit tests, to validate that the new configuration option works as expected and does not introduce regressions. All tests are currently passing.

```
[Wed Nov  1 12:53:42 PDT 2023] --------- Transmitting 27 items ---------
[Wed Nov  1 12:53:43 PDT 2023] Telemetry transmitted in 331.926261ms
[Wed Nov  1 12:53:43 PDT 2023] Response: 200
[Wed Nov  1 12:53:43 PDT 2023] Items accepted/received: 27/27
[Wed Nov  1 12:53:53 PDT 2023] --------- Transmitting 30 items ---------
[Wed Nov  1 12:53:53 PDT 2023] Telemetry transmitted in 73.171392ms
[Wed Nov  1 12:53:53 PDT 2023] Response: 200
[Wed Nov  1 12:53:53 PDT 2023] Items accepted/received: 30/30
[Wed Nov  1 12:54:04 PDT 2023] --------- Transmitting 27 items ---------
[Wed Nov  1 12:54:04 PDT 2023] Telemetry transmitted in 68.037724ms
[Wed Nov  1 12:54:04 PDT 2023] Response: 200
[Wed Nov  1 12:54:04 PDT 2023] Items accepted/received: 27/27
```

**Documentation:** <Describe the documentation added.>

TODO, in a follow up PR.

### CODE REVIEW

File: .chloggen/add-connection-string-azure-monitor-exporter.yaml
Comment (cijothomas) [inline .chloggen/add-connection-string-azure-monitor-exporter.yaml:18]: lets be more specific here. State that support for ConnectionString is added, and instrumentationkey approach is still supported for back-compat.

**Comment by cijothomas** [comment]
> Documentation Update in a follow up PR

Could you do that in this PR itself, so it is easy to see the new flow (for the non Golang folks to also help review!)

**Review by pcwiese — APPROVED** [review]
Looks good to me, apart from any missing documentation changes.

**Review by cijothomas — APPROVED** [review]
Thanks! looks good

**Comment by rajkumar-rangaraj** [comment]
@collector-contrib-approvers Could you please help merge this PR, got an approval from the code owner.

**Comment by rajkumar-rangaraj** [comment]
@open-telemetry/collector-contrib-approvers Could you please help merge this PR, got an approval from the code owner.

**Review by djaglowski — APPROVED** [review]

**Pull request merged**
